domain/model/service/modelservice.go

```diff
@@ -24,37 +26,41 @@ type ModelState interface {
 	Model(context.Context) (coremodel.ReadOnlyModel, error)
 }
 
-// ModelGetterState represents the state required for reading all model
-// information.
-type ModelGetterState interface {
+// ControllerState is the controller state required by this service. This is the
+// controller database, not the model state.
+type ControllerState interface {
+	// GetModel returns the model with the given UUID.
 	GetModel(context.Context, coremodel.UUID) (coremodel.Model, error)
+
+	// GetModelState returns the model state for the given model.
```

Comment:
As per discussion at the sprint we should document that we expect this func to return [modelerrors.NotFound] when no model exists for the UUID.